pkg/federate/kubefed/resource_test.go

```diff
@@ -0,0 +1,372 @@
+package kubefed
+
+import (
+	"context"
+	"errors"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	ctlutil "sigs.k8s.io/kubefed/pkg/controller/util"
+	kubefedopt "sigs.k8s.io/kubefed/pkg/kubefedctl/options"
+)
+
+type fakeClientWithUpdateError struct {
+	client.Client
+	msg string
+}
+
+var _ client.Client = &fakeClientWithUpdateError{}
```

Comment:
This is done in general to make the compiler check that a struct implements an interface.  It isn't really needed in this case since `fakeClientWithUpdateError` is used as a `client.Client` in code so the compiler will check anyway.